### PR TITLE
[Merged by Bors] - chore(Algebra.Module.LocalizedModule): use `IsLocalization` instead of `Localization`

### PR DESCRIPTION
**Pull request by xroblot**
The construction of `LocalizedModule` is done using `IsLocalization` rather than `Localization`. The corresponding instances for `Localization` are deduced automatically by Lean. One drawback is that many instances are now marked `noncomputable`. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/LocalizedModule.lean [hunk 1]
```diff
@@ -415,43 +412,46 @@ theorem mk_cancel_common_right (s s' : S) (m : M) : mk (s' • m) (s * s') = mk
...
-      obtain ⟨y, s, rfl : IsLocalization.mk' _ y s = r⟩ := IsLocalization.mk'_surjective S r
-      obtain ⟨⟨a₁, s₁⟩, rfl : mk a₁ s₁ = x₁⟩ := Quotient.exists_rep x₁
-      obtain ⟨⟨a₂, s₂⟩, rfl : mk a₂ s₂ = x₂⟩ := Quotient.exists_rep x₂
-      rw [mk_mul_mk, ← Localization.mk_eq_mk', mk_smul_mk, mk_smul_mk, mk_mul_mk, mul_left_comm,
-        mul_smul_comm])
+variable (T)
 
-theorem algebraMap_mk {A : Type*} [Semiring A] [Algebra R A] (a : R) (s : S) :
-    algebraMap _ _ (Localization.mk a s) = mk (algebraMap R A a) s := by
+noncomputable instance {A : Type*} [Semiring A] [Algebra R A] :
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Module/LocalizedModule.lean:425]: Can you prove the two properties in separate lemmas, so we can have this definition in term mode?

Comment (xroblot) [reply Mathlib/Algebra/Module/LocalizedModule.lean]: Is this what you had in mind?

Comment (riccardobrasca) [reply Mathlib/Algebra/Module/LocalizedModule.lean]: Yes, thanks.

**Comment by erdOne** [comment]
I wonder if this should be done with `IsLocalizedModule`?

**Comment by erdOne** [comment]
In #9210 Mathlib/RingTheory/Localization/Module.lean imports Mathlib/Algebra/Module/LocalizedModule.lean instead, which IMO is the right direction. What in that file does this PR need?

**Comment by erdOne** [comment]
Oh I see. We should probably find a better home for `IsLocalizedModule.isBaseChange` then.

**Comment by xroblot** [comment]
> Oh I see. We should probably find a better home for `IsLocalizedModule.isBaseChange` then.

Yes, as you can see most of the heavy lifting is done in #9210. This PR is just about replacing `Localization` by `IsLocalization`.

**Comment by riccardobrasca** [comment]
This is strictly more general than the current situation, right?

**Comment by xroblot** [comment]
> This is strictly more general than the current situation, right?

Yes, the result about `Localization` can be recovered since we have the instance `Localization.isLocalization`. In particular, Lean can automatically infer the instance about `Localization` from the ones about `IsLocalization`.

**Comment by riccardobrasca** [comment]
> > This is strictly more general than the current situation, right?
> 
> Yes, the result about `Localization` can be recovered since we have the instance `Localization.isLocalization`. In particular, Lean can automatically infer the instance about `Localization` from the ones about `IsLocalization`.

Thanks, let's merge this and maybe improve things later.

bors merge

**Pull request closed without merging**